### PR TITLE
Adds Hippie icon check

### LINKED ISSUE #4964: Brass Chairs have no sprites
In-Game report from critiasblue:
Server info: 24552 (Hippie Station)

>General description of the issue
Brass chairs have no sprite, making them invisible

>What you expected to happen
I built a brass chair out of brass and expected a chair to appear

>What actually happened
No chair appeared and I had to right click the ground to see if it appeared

>Steps to reproduce if possible
Make a brass chair

### PR DESCRIPTION
**Pull request by JohnGinnane**
This is something that's been on my mind for a while now. The current problem is that if we want to revert any tg icons (e.g. chairs or computers), we create a copy of the .dmi under the HippieStation folder and then do what we want. If tg adds icons to their file (new brass chair), and we don't reflect this change somehow then there's a chance that we end up with invisible objects

There's a lot of files changed because we're overriding a lot of icons, so I've broken this up into 3 separate commits so it's easier to follow:

1. Creating a new variable and proc that checks if icon state exists in the hippie icon file
2. Fixed some icons not being modularised
3. Replaced all instances of us overriding the `icon` var with us setting the `icon_hippie` var, which is used in the proc created in the first commit

If `icon`, `icon_state`, and `icon_hippie` are all defined then we check if `icon_state` exists in `icon_hippie` file. If it does then we set `icon = icon_hippie` and so we end up with our sprites

Here you can see what we now do **if we want to override tg's icons**
image

Here's the result:
image

Just so you know, here's the search I used to replace all those files:
image

:cl: JohnGinnane
fix: Most, if not all, invisible icons caused by our doing
code: Added a new proc which verifies if the hippie icon is there before overriding tg's icon
/:cl:

Fixes #4964

A re-do of #7228 seeing as a botched it

### CODE REVIEW

**Comment by JohnGinnane** [comment]
@Carbonhell this is ready

File: hippiestation/code/modules/surgery/tools.dm [hunk 1]
```diff
@@ -1,28 +1,28 @@
 /obj/item/retractor/syndicate
 	name = "syndicate retractor"
 	desc = "Reinforced with carbon-plasma alloy for a stalwart grip."
-	icon = 'hippiestation/icons/obj/surgery.dmi'
+	icon_hippie = 'hippiestation/icons/obj/surgery.dmi'
```
Comment (tortellinitony) [inline hippiestation/code/modules/surgery/tools.dm]: not necessary

File: hippiestation/code/modules/surgery/organs/tails.dm [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /obj/item/organ/tail/cat/tcat
-	icon = 'hippiestation/icons/obj/surgery.dmi'
+	icon_hippie = 'hippiestation/icons/obj/surgery.dmi'
```
Comment (tortellinitony) [inline hippiestation/code/modules/surgery/organs/tails.dm]: not necessary

File: hippiestation/code/modules/surgery/organs/ears.dm [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /obj/item/organ/ears/cat/tcat
-	icon = 'hippiestation/icons/obj/surgery.dmi'
+	icon_hippie = 'hippiestation/icons/obj/surgery.dmi'
```
Comment (tortellinitony) [inline hippiestation/code/modules/surgery/organs/ears.dm]: not necessary

File: hippiestation/code/modules/surgery/organs/butts.dm [hunk 4]
```diff
@@ -2,7 +2,7 @@
 	name = "butt"
 	desc = "extremely treasured body part"
 	alternate_worn_icon = 'hippiestation/icons/mob/head.dmi'
-	icon = 'hippiestation/icons/obj/butts.dmi'
+	icon_hippie = 'hippiestation/icons/obj/butts.dmi'
```
Comment (tortellinitony) [inline hippiestation/code/modules/surgery/organs/butts.dm]: not necessary

File: hippiestation/code/modules/sports/hockey.dm [hunk 5]
```diff
@@ -4,7 +4,7 @@
 /obj/item/hockeypack
 	name = "Ka-Nada Special Sport Forces Hockey Pack"
 	desc = "Holds and powers a Ka-Nada SSF Hockey Stick, A devastating weapon capable of knocking men around like toys and batting objects at deadly velocities."
-	icon = 'hippiestation/icons/obj/clothing/back.dmi'
+	icon_hippie = 'hippiestation/icons/obj/clothing/back.dmi'
```
Comment (tortellinitony) [inline hippiestation/code/modules/sports/hockey.dm]: not necessary

File: hippiestation/code/modules/sports/bowling.dm [hunk 6]
```diff
@@ -23,7 +23,7 @@
 
 /obj/item/bowling
 	name = "bowling ball"
-	icon = 'hippiestation/icons/obj/items_and_weapons.dmi'
+	icon_hippie = 'hippiestation/icons/obj/items_and_weapons.dmi'
```
Comment (tortellinitony) [inline hippiestation/code/modules/sports/bowling.dm]: not necessary

File: hippiestation/code/modules/reagents/reagent_containers/solid_reagent.dm [hunk 7]
```diff
@@ -1,7 +1,7 @@
 /obj/item/reagent_containers/food/snacks/solid_reagent
 	name = "solidified chemicals"
 	desc = "Are you sure eating this is a good idea?"
-	icon = 'hippiestation/icons/obj/chemical.dmi'
+	icon_hippie = 'hippiestation/icons/obj/chemical.dmi'
```
Comment (tortellinitony) [inline hippiestation/code/modules/reagents/reagent_containers/solid_reagent.dm]: not necessary

File: hippiestation/code/modules/reagents/chemistry/machinery/lcasschem.dm [hunk 8]
```diff
@@ -1,7 +1,7 @@
 /obj/machinery/chem
 	density = TRUE
 	anchored = TRUE
-	icon = 'hippiestation/icons/obj/chemical.dmi'
+	icon_hippie = 'hippiestation/icons/obj/chemical.dmi'
```
Comment (tortellinitony) [inline hippiestation/code/modules/reagents/chemistry/machinery/lcasschem.dm]: not necessary

File: hippiestation/code/modules/projectiles/guns/magic/staffofrevenant.dm [hunk 9]
```diff
@@ -5,7 +5,7 @@
 	ammo_type = /obj/item/ammo_casing/magic/staffofrevenant
 	icon_state = "staffofrevenant"
 	item_state = "staffofrevenant"
-	icon = 'hippiestation/icons/obj/items_and_weapons.dmi'
+	icon_hippie = 'hippiestation/icons/obj/items_and_weapons.dmi'
```
Comment (tortellinitony) [inline hippiestation/code/modules/projectiles/guns/magic/staffofrevenant.dm]: not necessary

File: hippiestation/code/modules/projectiles/guns/grenade_launcher.dm [hunk 10]
```diff
@@ -1,5 +1,5 @@
 /obj/item/gun/ballistic/revolver/grenadelauncher/mgl
-	icon = 'hippiestation/icons/obj/guns/projectile.dmi'
+	icon_hippie = 'hippiestation/icons/obj/guns/projectile.dmi'
```
Comment (tortellinitony) [inline hippiestation/code/modules/projectiles/guns/grenade_launcher.dm]: not necessary

**Review by tortellinitony — DISMISSED** [review]

**Comment by tortellinitony** [comment]
Go through and make sure the icons you switched are actually overriding something

File: hippiestation/code/modules/mob/dead/new_player/sprite_accessories.dm [hunk 11]
```diff
@@ -1,24 +1,49 @@
+/datum/sprite_accessory
+    var/icon_hippie
+
+/datum/sprite_accessory/proc/check_hippie_icon()
```
Comment (Carbonhell) [inline hippiestation/code/modules/mob/dead/new_player/sprite_accessories.dm]: Ugly copypaste, can't you just make this proc a /datum proc so both this and atoms own it

File: hippiestation/code/game/turfs/turf.dm [hunk 12]
```diff
@@ -16,4 +16,9 @@
 
 			for (var/obj/item/stack/rods/R in H.contents)
 				if (R.pinned)
-					R.pinned = null
\ No newline at end of file
+					R.pinned = null
+
+/turf/Initialize()
```
Comment (Carbonhell) [inline hippiestation/code/game/turfs/turf.dm:21]: ??? Turfs are atoms, why is this needed

**Review by Carbonhell — DISMISSED** [review]

File: hippiestation/code/modules/mob/dead/new_player/sprite_accessories.dm (hunk 11 above)
Comment (JohnGinnane) [reply hippiestation/code/modules/mob/dead/new_player/sprite_accessories.dm]: I should do that

File: hippiestation/code/game/turfs/turf.dm (hunk 12 above)
Comment (JohnGinnane) [reply hippiestation/code/game/turfs/turf.dm:21]: I was surprised by this but it seems to be necessary because when I was testing this last night I noticed the walls and reinf walls were weird, turns out they were still using tg icons. I checked documentation as I thought turfs were atoms too but idk why this is necessary. Possibly because turf's Initialize() is different to atom's Initialize()?

File: hippiestation/code/modules/mob/dead/new_player/sprite_accessories.dm (hunk 11 above)
Comment (JohnGinnane) [reply hippiestation/code/modules/mob/dead/new_player/sprite_accessories.dm]: turns out those files don't even need it

**Comment by JohnGinnane** [comment]
One of the fodlers was acting up:

https://github.com/HippieStation/HippieStation/tree/master/code/game/objects/effects/decals/Cleanable

It kept messing up the dme by changing the reference to and from capital C so I've just renamed the folder to fix it, the files themselves are unchanged

File: hippiestation/code/game/turfs/turf.dm (hunk 12 above)
Comment (tortellinitony) [reply hippiestation/code/game/turfs/turf.dm:21]: Perhaps it would be better to have the icon change AFTER The turf has initialized

**Comment by tortellinitony** [comment]
You didn't make it a subsystem!

File: hippiestation/code/game/turfs/turf.dm (hunk 12 above)
Comment (JohnGinnane) [reply hippiestation/code/game/turfs/turf.dm:21]: Oops

**Comment by JohnGinnane** [comment]
:(

image

**Comment by JohnGinnane** [comment]
@tortellinitony @Carbonhell pls you're my only hope

also this fixes one of our folders being incorrectly capitalised and that's messing with other people's PRs

File: hippiestation/code/game/turfs/turf.dm (hunk 12 above)
Comment (Carbonhell) [reply hippiestation/code/game/turfs/turf.dm:21]: i still don't get why this is needed, if the ..() is the problem just do . = ..() then call check hippie icon after it no?

**Review by Carbonhell — APPROVED** [review]
